### ISSUE TITLE
plugins migration not run for custom plugins installed outside kong's default installation path

### ISSUE THREAD

**Issue by ms5838** [Kong/kong#3394]
### Summary
Migrations up does not recognize and execute the "up" command for a new custom plugin that uses DAO and is being installed in external folder location. The same plugin if installed in Kong's default installation path works fine.

I see a related old issue but it seems the issue was fixed 0.9. We are using 0.11.2 and experiencing this behavior with that version

Old similar issue: https://github.com/Kong/kong/issues/1708 

### Steps To Reproduce

1. Create a custom plugin that uses DAO and install it in a folder external to Kong's default installation.
    example: /home/janedoe/kong/plugins/myplugin
2. Add the plugin name to custom_plugins setting in kong.conf
3. Add path to external folder to lua_package_path in kong.conf
    example: lua_package_path = /home/janedoe/?.lua;;
4. Restart kong to see that it gives "Please run `kong migrations up` to update/initialize the database schema." error.
5. Run "kong migrations up"
6. kong migrations up finishes without error, but there's no migration that gets executed for the custom plugin. There are no errors in error.log
7. Try to start kong again and see kong complains with "Please run `kong migrations up` to update/initialize the database schema." error.

Workaround: install the custom plugin in kong's default installation folder kong/luarocks/share/lua/5.1/kong/plugins
We need the ability to put the custom plugins in external folder and hence workaround is not feasible.

Am I missing anything? or did the issue reported in the older version resurfaced back again in 0.11.2?

### Additional Details & Logs

- Kong version 0.11.2
- Kong error logs: No errors in error.log
- Operating System: Linux xxxxxxxx 2.6.32-696.10.3.el6.x86_64 #1 SMP Thu Sep 21 12:12:50 EDT 2017 x86_64 x86_64 x86_64 GNU/Linux

**Comment by thibaultcha** [comment]
Hi,

I just ran into this: as a short term solution, you should set your `LUA_PATH` environment variable to point to your custom plugin's installation path (instead of `lua_package_path` in kong.conf or `KONG_LUA_PACKAGE_PATH`).

E.g. I just moved:
```
export KONG_LUA_PACKAGE_PATH="/path/to/some-custom-plugin/?.lua;;"
```

to:
```
export LUA_PATH="/path/to/some-custom-plugin/?.lua;$LUA_PATH"
```

And ran `kong migrations up` again, which successfully ran the plugin's migration.

That is because `lua_package_path` is only for runtime Kong (within nginx), and is not being considered by the CLI (the `kong migrations` command)

**Comment by ms5838** [comment]
@thibaultcha Thanks for the immediate response, really appreciate it. 

This worked, but I had to perform one additional step. 

If I just commented the lua_package_path in kong.conf and exported LUA_PATH as you mentioned, it started giving me this error:
./kong:3: module 'luarocks.loader' not found:

The shell from which I am executing kong migrations up never had LUA_PATH set, even before. It appears luarocks internally creates LUA_PATH if it is not already set in the shell.

So the additional step I had to perform was to get the current LUA_PATH that's set internally by running "luarocks path" command and then take the result of that, and append my custom login plugin path to it and then export that LUA_PATH to get this to work.

You mentioned this as a short term solution. Will this be addressed in the longer term to reference lua_package_path in CLI as well? Thanks!

**Comment by thibaultcha** [comment]
> You mentioned this as a short term solution. Will this be addressed in the longer term to reference lua_package_path in CLI as well? Thanks!

Yes, we can certainly do this, although we would welcome a PR for this, or else we are working on some other priorities internally and might need time to address this.

Thanks for reporting!

**Comment by cewood** [comment]
Hi,
Just wanted to add that I also ran into this issue just now. I tried the suggestion above, but this didn't work.

Is there any ETA on when this is likely to be worked on?

Thanks!

**Comment by yasn77** [comment]
I think this could be fixed by https://github.com/Kong/kong/pull/5509

**Comment by cewood** [comment]
> I think this could be fixed by #5509

Indeed, that looks spot on! I was trying to find that line in the code a littler earlier :see_no_evil:

**Comment by bungle** [comment]
It looks like this got fixed already. Please reopen if that was not the case.
